gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/ApiReactorHandlerFactory.java

```diff
@@ -217,14 +208,34 @@ public ReactorHandler create(Api api) {
         return null;
     }
 
-    protected ExecutionContextFactory executionContextFactory(ComponentProvider componentProvider) {
-        return new ExecutionContextFactory(componentProvider);
+    private boolean isV3ExecutionMode(Api api) {
+        return api.getExecutionMode() == null || api.getExecutionMode() == ExecutionMode.V3;
+    }
+
+    protected <T extends AbstractExecutionContextFactory> T executionContextFactory(
```

Comment:
should it be a private method ?

Reply:
This is because there is currently a DebugExecutionContextFactory that needs to override it. Don't know if it will stay as it but was necessary at this stage.